Catappult/appcoins-wallet-android
# [APPC-4659] Preselected payment methods payment fields for purchase header

**What does this PR do?**

   Add call to get sku details. Also created payment methods to have the individual logic for each payment method.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] ProductInfoData.kt
- [ ] GetProductInfoUseCase.kt
- [ ] PaymentManager.kt
- [ ] MainFragment.kt
- [ ] MainFragmentUiState.kt
- [ ] MainViewModel.kt
- [ ] GetWalletInfoUseCase.kt
- [ ] SDKUriParserImpl.kt
- [ ] PaymentMethodsFragment.kt
- [ ] PaymentMethodsUiState.kt
- [ ] PaymentMethodsViewModel.kt
- [ ] IAPPaymentMethodScreen.kt
- [ ] PaymentActivity.kt
- [ ] PaymentManagerFactory.kt
- [ ] IAPModule.kt
- [ ] APPCFactory.kt
- [ ] AskSomeoneToPayFactory.kt
- [ ] CreditCardFactory.kt
- [ ] PayPalV1Factory.kt
- [ ] PayPalV2Factory.kt
- [ ] VKFactory.kt
- [ ] APPCPaymentMethod.kt
- [ ] AskSomeoneToPayPaymentMethod.kt
- [ ] CreditCardPaymentMethod.kt
- [ ] PayPalV1PaymentMethod.kt
- [ ] PayPalV2PaymentMethod.kt
- [ ] VKPaymentMethod.kt
- [ ] PaymentMethod.kt
- [ ] PaymentMethodCreator.kt
- [ ] PaymentMethodFactory.kt
- [ ] PaymentMethodEntity.kt
- [ ] WalletInfo.kt
- [ ] Composables.kt
- [ ] iab_graph.xml

**How should this be manually tested?**

  Start an IAP flow. Check the payment methods list. check the price changes and validate if its correct.

**What are the relevant tickets?**

  Tickets related to this pull-request: [APPC-4659](https://aptoide.atlassian.net/browse/APPC-4659)

**Code Review Checklist**



[APPC-4659]: https://aptoide.atlassian.net/browse/APPC-4659?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
